src/WaterPokemon.java

```diff
@@ -0,0 +1,114 @@
+                damage = 10;
+                break;
+            case "grass":
+                damage = 15;
+                break;
+            default:
+                damage = 5;
+                break;
+        }
+        enemy.takeDamage(damage);
+        System.out.println(enemy.getName() + " loses " + damage + " hp");
+        System.out.println("Remaining hp of " + enemy.getName() + " is " + enemy.getHp());
+    }
+
+
+    void hydroPump(Pokemon name, Pokemon enemy) {
+        System.out.println(this.getName() + " attacks " + enemy.getName() + " with hydroPump");
+
+        int damage;
+        switch (enemy.getType()) {
+            case "fire":
+                damage = 20;
+                break;
+            case "electric":
+                damage = 10;
+                break;
+            case "grass":
+                damage = 15;
+                break;
+            default:
+                damage = 5;
+                break;
+        }
+        enemy.takeDamage(damage);
+        System.out.println(enemy.getName() + " loses " + damage + " hp");
+        System.out.println("Remaining hp of " + enemy.getName() + " is " + enemy.getHp());
+    }
+
+
+    void rainDance(Pokemon name, Pokemon enemy) {
+        System.out.println(this.getName() + " attacks " + enemy.getName() + " with rainDance");
+
+        if (enemy.getType().equals("electric")) {
+            System.out.println("RainDance has no effect on " + enemy.getName());
+        } else {
+            int damage;
+            switch (enemy.getType()) {
+                case "fire":
+                    damage = 20;
+                    break;
+                case "grass":
+                    damage = -10;
+                    break;
+                default:
+                    damage = 5;
+                    break;
+            }
+            enemy.takeDamage(damage);
+            System.out.println(enemy.getName() + " loses " + Math.abs(damage) + " hp");
+            System.out.println("Remaining hp of " + enemy.getName() + " is " + enemy.getHp());
+        }
+    }
+
+    void surf(Pokemon name, Pokemon enemy) {
+        System.out.println(this.getName() + " attacks " + enemy.getName() + " with surf");
+
+        if (enemy.getType().equals("electric")) {
+            System.out.println("RainDance has no effect on " + enemy.getName());
+        } else {
+            int damage;
+            switch (enemy.getType()) {
+                case "fire":
+                    damage = 20;
+                    break;
+                case "grass":
+                    damage = -10;
+                    break;
+                default:
+                    damage = 5;
+                    break;
+            }
+            enemy.takeDamage(damage);
+            System.out.println(enemy.getName() + " loses " + Math.abs(damage) + " hp");
+            System.out.println("Remaining hp of " + enemy.getName() + " is " + enemy.getHp());
+        }
+    }
+
+    public void attack() {
```

Comment:
Wordt deze methode gebruikt? Het is misschien beter om dit in de super klasse te definieren, misschien als abstract, want deze methode staat nu in elke subklasse